hide_invoice_shipping_address/models/account_move.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2024 Roger Sans <roger.sans@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    invoice_shipping_address_print = fields.Boolean(
+        string='Print The Shipping Invoice On The Invoice',
+        compute='_get_invoice_shipping_address_value',
+        store=True,
+    )
+
+    @api.depends('partner_shipping_id.invoice_shipping_address_print',
+                 'partner_id.invoice_shipping_address_print',
+                 'company_id.invoice_shipping_address_print',)
+    def _get_invoice_shipping_address_value(self):
```

Comment:
This method needs to be rewritten, as it doesn't do what it is supposed to do. The hierarchy to determine the value for invoice_shipping_address_print in account.move is as follows:

1. If partner_shipping_id has a value set in its field invoice_shipping_address_print, that will be the value set in the field invoice_shipping_address_print in account.move. If no value is set, check point 2.
2. If the commercial partner (commercial_partner_id) of the partner_shipping_id  has a value set in its field invoice_shipping_address_print, that will be the value set in the field invoice_shipping_address_print in account.move. If no value is set, check point 3.
3. The value set in the field invoice_shipping_address_print in account.move is the value set in the general settings.